numpyro/infer/svi.py

```diff
@@ -86,33 +86,127 @@ class SVI(object):
 
     .. doctest::
 
-        >>> from jax import random
         >>> import jax.numpy as jnp
+        >>> from jax import random
+
         >>> import numpyro
         >>> import numpyro.distributions as dist
-        >>> from numpyro.distributions import constraints
-        >>> from numpyro.infer import Predictive, SVI, Trace_ELBO
-
-        >>> def model(data):
-        ...     f = numpyro.sample("latent_fairness", dist.Beta(10, 10))
-        ...     with numpyro.plate("N", data.shape[0]):
-        ...         numpyro.sample("obs", dist.Bernoulli(f), obs=data)
-
-        >>> def guide(data):
-        ...     alpha_q = numpyro.param("alpha_q", 15., constraint=constraints.positive)
-        ...     beta_q = numpyro.param("beta_q", lambda rng_key: random.exponential(rng_key),
-        ...                            constraint=constraints.positive)
-        ...     numpyro.sample("latent_fairness", dist.Beta(alpha_q, beta_q))
-
-        >>> data = jnp.concatenate([jnp.ones(6), jnp.zeros(4)])
-        >>> optimizer = numpyro.optim.Adam(step_size=0.0005)
-        >>> svi = SVI(model, guide, optimizer, loss=Trace_ELBO())
-        >>> svi_result = svi.run(random.PRNGKey(0), 2000, data)
+        >>> import numpyro.optim as optim
+        >>> from numpyro.infer import SVI, Trace_ELBO, Predictive
+        >>> from numpyro.infer.autoguide import AutoNormal
+
+        >>> # Modeling person's weight from a 10-observation snippet
+        >>> # of 'Howell1' dataset from Statistical Rethinking 2nd ed. (R code 5.45)
```

Comment:
I think it is better to use the current example, which is concise, focuses on SVI inference (not modeling), and illustrates different aspects like initializing parameters. You can move this example to a standalone example/tutorial in https://github.com/pyro-ppl/numpyro/tree/master/examples .

Reply:
For the current docstring, I think we are missing prior/posterior predictive?

Reply:
I wouldn't call the current doc example accessible to outsiders (compared to sklearn examples or compared to Pyro even: e.g. no `alpha0`, `beta0` variables from [reference](http://pyro.ai/examples/svi_part_i.html)). but my proposal is not very laconic i admit and i'm not the maintainer.

so yeah, if you want i can just secure all the 4 sampling showcase and channel the rest into examples section.

also sorry for being dilatory. i'm not abandoning conrtib efforts, i just have to do a lot of crisis management due to what's happening.

Reply:
I see. I guess a beta binomial model is not a good starting example. How about using a simple linear regression with example with scalar parameters instead? Like learning `y = w * x + b` where `x = jnp.arange(10)`, `y = 2 * x + 1 + random.normal(random.PRNGKey(0), shape=x.shape) * 0.1`. I think we need custom guide so it is better to avoid auto guide here. We also need `numpyro.param(..., lambda rng_key: ..., ...)` to initialize parameters.

Moving your example to the examples section sounds great to me. Please take your time. We have several PRs that take years to finish. ;) But we might occasionally check to see if you want to make those enhancements available for the next releases.

Reply:
Haha thanks @fehiepsi 

@fehiepsi I've went through "SVI part 1 " Pyro example and decided that it's actually fine, just needs some more clarity in Numpyro adaptation. So I've tried to make it more coherent with Pyro implementation at the end of the tutorial so that it'd be easier to refer and compare differences of Numpyro SVI API to Pyro SVI API. Please check out the last commits

Reply:
@ColdTeapot273K would you mind if I push a commit to revise the CL and merge this?